### DISCUSSION TITLE
How to delete note using backend API?

### DISCUSSION THREAD

**Discussion by WPum** [TriliumNext/Trilium#2792] [Q&A] [answered]
`toggleNoteInParent(false, XX, XX)` doesn't work. I got the following error message when I try to delete note with it:
```
Cannot remove branch XX between child XX and parent XX because this would delete the note as well.
```

**Comment by zadam** [comment, marked as answer]
Hi, see https://zadam.github.io/trilium/backend_api/AbstractEntity.html

```
note.markAsDeleted();
```

**Reply by WPum** [reply]
> Hi, see https://zadam.github.io/trilium/backend_api/AbstractEntity.html
> 
> ```
> note.markAsDeleted();
> ```

Thanks for your reply. Calling `note.markAsDeleted()` will cause Trilium to crash. After restarting the server, there're some consistency issues in the database. Should I do some other cleanup works?

**Reply by zadam** [reply]
Hi, can you please share some details on the crash? Log/errors ...

**Reply by WPum** [reply]
Code: `note.markAsDeleted()`

The following error message is given in the backend log:
```
ERROR: Cannot find note XX to replace image link.
ERROR: Execution of script "App" (XX) failed with error: Load of script note "App" (XX) failed with: Cannot read property 'isTemplate' of undefined
```

After a reload, the log shows:
```
Consistency issue fixed: Branch XX has been deleted since associated note XX is deleted.
Consistency issue fixed: Removed attribute XX because owning note XX is also deleted.
All consistency checks passed after some fixes (took 20ms)
```

And more...
```
ERROR: Listener threw error: TypeError: Cannot read property 'getRelations' of undefined
    at runAttachedRelations (C:\trilium-0.50.3\src\services\handlers.js:11:14)
    at C:\trilium-0.50.3\src\services\handlers.js:41:9
    at Object.emit (C:\trilium-0.50.3\src\services\events.js:49:17)
    at Attribute.markAsDeleted (C:\trilium-0.50.3\src\becca\entities\abstract_entity.js:124:22)
    at C:\trilium-0.50.3\src\services\consistency_checks.js:464:31
    at Function.<anonymous> (C:\trilium-0.50.3\src\services\consistency_checks.js:31:41)
    at Function.sqliteTransaction (C:\trilium-0.50.3\node_modules\better-sqlite3\lib\methods\transaction.js:65:24)
    at Object.transactional (C:\trilium-0.50.3\src\services\sql.js:236:52)
    at ConsistencyChecks.findAndFixIssues (C:\trilium-0.50.3\src\services\consistency_checks.js:31:21)
    at ConsistencyChecks.findLogicIssues (C:\trilium-0.50.3\src\services\consistency_checks.js:454:14)
```

**Reply by zadam** [reply]
Hi, I realized that `note.markAsDeleted()` is probably too low-level for scripting since it doesn't take care of things like attributes (which needs to be deleted in cascade) etc.

I added new methods which should work as expected:

```
note.deleteNote();
```

https://zadam.github.io/trilium/backend_api/Note.html

This will be released soon in 0.51.1

**Reply by WPum** [reply]
> Hi, I realized that `note.markAsDeleted()` is probably too low-level for scripting since it doesn't take care of things like attributes (which needs to be deleted in cascade) etc.
> 
> I added new methods which should work as expected:
> 
> ```
> note.deleteNote();
> ```
> 
> https://zadam.github.io/trilium/backend_api/Note.html
> 
> This will be released soon in 0.51.1

OK, thanks for that.
